app/models/user.rb

```diff
@@ -10,6 +10,27 @@ class User < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
 
+  has_many :confirmed_friendships, -> { where confirmed: true }, class_name: 'Friendship'
+  has_many :friends, through: :confirmed_friendships
+
+  # Users who needs to confirm friendship
+  has_many :pending_friendships, -> { where confirmed: false }, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :pending_friends, through: :pending_friendships, source: :friend
+
+  # Users who requested to be friends (needed for notifications)
+  has_many :inverted_friendships, -> { where confirmed: false }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friend_requests, through: :inverted_friendships, source: :friend
+#rubocop:disable all
+  def friends_and_own_posts
+    Post.where(user: (self.friends.to_a << self))
+    # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
+  end
+
```

Comment:
- [ ] As mentioned in the last review, currently, users who are not friends can all view all users' posts even when they are not friends. The expected behavior is user A can only view their own posts as well as their friends' posts (say user B). Please, let's fix it.